python/tests/utils/test_schema_conversion.py

```diff
@@ -229,3 +234,127 @@ def test_avro_list_required_record():
     iceberg_schema = AvroSchemaConversion().avro_to_iceberg(avro_schema)
 
     assert expected_iceberg_schema == iceberg_schema
+
+
+def test_resolve_union():
```

Comment:
Should these conversion test cases go in a separate PR? It doesn't seem related to the reader.

Reply:
Yes, let me split those out

Reply:
https://github.com/apache/iceberg/pull/4972